mantidproject/mantid
# Create alternative method for energy transfer by sample changer state

**Description of work.**

This PR add the option to group the outputs of ISISEnergy transfer by sample changer position

**To test:**

1. Open Indirect Data reduction
2. Load Runs 94297-94312
3. Under output options select Group by Sample
4. Click run
5. check that outputs are grouped into 3 group workspaces each labeled as either top, mid or btm
6. Select one of the workspaces in the groups and show sample logs,
7. the SAMP_POS log should be 6.0 for btm workspaces, 85.5 for mid and 165.0 for top
8. repeat for runs 32500-32510
9. These should be in a single group labeled none, and workspaces should not have SAMP_POS in their logs.

Fixes #33531

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
